test/e2e/framework/util.go

```diff
@@ -459,14 +459,14 @@ func NodeOSDistroIs(supportedNodeOsDistros ...string) bool {
 	return false
 }
 
-func ProxyMode(f *Framework) (string, error) {
+func ProxyMode(f *Framework, hostNetwork bool) (string, error) {
```

Comment:
Will this function work if not in hostNetwork?

It seems to me that it is relying on hitting the host at a given port via localhost, which won't resolve the same when run in hostNetwork vs pod networked.

Reply:
Apparently so. The only place this is used in, is in the test ``should preserve source pod IP for traffic thru service cluster IP``, which is even marked as passed in the ``pull-kubernetes-e2e-kops-aws`` and the ``pull-kubernetes-e2e-gce`` jobs.

Reply:
Something is fishy about that. This is exec'ing into a pod and curling a localhost URL of a sleep pod.

https://github.com/kubernetes/kubernetes/blob/8616687c908d03f5bfdbb7cef03468d232c00132/test/e2e/framework/util.go#L3431
https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/test/e2e/network/networking.go#L87
https://github.com/kubernetes/kubernetes/blob/ef195428a0c5b16e3f3621b7fe59fa151ca28929/cmd/kube-proxy/app/server.go#L507